### PR TITLE
Automated builds 2

### PR DESCRIPTION
**Pull request by sharkwouter**
This is based on the most recent code I have. It successfully builds on Ubuntu and Windows. Builds are made for each release.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,46 +5,69 @@ project(match-theory VERSION 0.1.0)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-add_compile_options(-Wall -Wextra -pedantic)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```
Comment (protheory8) [inline CMakeLists.txt:8]: Does this also match against Clang?

File: CMakeLists.txt [hunk 2]
```diff
@@ -59,32 +82,97 @@ if (VITA)
...
+          ${JSONCPP_INCLUDE_DIRS}
+     )
+     if(NOT CMAKE_BUILD_TYPE MATCHES "Release")
+         set(CMAKE_BUILD_TYPE "Debug")
+     endif()
+     
      file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/assets
-               DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+          DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/)
 ELSE()
```
Comment (protheory8) [inline CMakeLists.txt]: `ELSE()` and `ENDIF()` probably should be written in lowercase for consistency, unless there is a difference between these two that I don't understand.

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,46 +5,69 @@ project(match-theory VERSION 0.1.0)
 set(CMAKE_CXX_STANDARD 17)
```
Comment (protheory8) [inline CMakeLists.txt:5]: Somehow I've never noticed this but this file uses 5 space indentation for some reason.

**Review by protheory8 — COMMENTED** [review]
Only one comment about something that isn't style

File: CMakeLists.txt (hunk 1 above)
Comment (sharkwouter) [reply CMakeLists.txt:8]: It does not. Does clang supports the same flags? I found that MSVC does not

File: CMakeLists.txt (hunk 3 above)
Comment (sharkwouter) [reply CMakeLists.txt:5]: Probably a copy and paste error. I'll address it.

**Pull request merged**